src/tools/rbd_mirror/Types.h

```diff
@@ -32,6 +32,15 @@ enum {
   l_rbd_mirror_snapshot_last,
 };
 
+// Labeled performance counters
+enum {
+  l_rbd_mirror_snapshot_remote_timestamp = l_rbd_mirror_snapshot_last,
```

Comment:
There are some overlapping values for `g_snapshot_perf_counters` and `m_perf_counters`, I tried adding them to the same enum but I had to add the image based keys to the global perf counters too to avoid the daemon to crash. The other way I tried adding the values was changing the range of just the `m_perf_counters` but that meant there was an extra key `l_rbd_mirror_snapshot_last` which also made the daemon crash at runtime.

I can introduce duplicate keys just for `m_perf_counters` if we can keep it in the same enum.

Reply:
I don't think it makes sense to have two versions of the same perf counter: an unlabeled one for the global value and a labeled one for the per-image value.  I would report the global value through the same labeled perf counter, with `""` label values.